datafusion/core/src/catalog/mod.rs

```diff
@@ -132,30 +129,3 @@ impl<'a> From<ResolvedTableReference<'a>> for TableReference<'a> {
         }
     }
 }
-
-impl<'a> TryFrom<&'a sqlparser::ast::ObjectName> for TableReference<'a> {
```

Comment:
We can't do this because we need to normalize the identifiers and that requires creating new strings and that didn't go down well with the defined lifetimes here, so I created a function in the SQL planner to do this conversion instead.

Reply:
cc @returnString